SocketRocket/SRWebSocket.m

```diff
@@ -110,8 +110,14 @@ @interface _SRRunLoopThread : NSThread
     assert(length >= 0);
     assert(length <= UINT32_MAX);
     CC_SHA1(bytes, (CC_LONG)length, md);
-    
-    return [[NSData dataWithBytes:md length:CC_SHA1_DIGEST_LENGTH] base64Encoding];
+
+    NSData *data = [NSData dataWithBytes:md length:CC_SHA1_DIGEST_LENGTH];
+    NSString *dataString;
+    if ([data respondsToSelector:@selector(base64EncodedStringWithOptions:)])
```

Comment:
To be defensive, please use braces around the body of your `if` and `else` statements; after all, that's how [goto fail;](https://www.imperialviolet.org/2014/02/22/applebug.html) happened!